datafusion/core/src/physical_optimizer/sort_enforcement2.rs

```diff
@@ -0,0 +1,2872 @@
+                    top_reversed_requirement,
+                    || child.equivalence_properties(),
+                ) {
+                    should_reverse_window_sort_requirements(
+                        child,
+                        top_requirement,
+                        top_reversed_requirement,
+                    )
+                } else {
+                    requirements_compatible(
+                        top_reversed_requirement,
+                        window_plan.required_input_ordering()[0].as_deref(),
+                        || window_plan.equivalence_properties(),
+                    ) || requirements_compatible(
+                        window_plan.required_input_ordering()[0].as_deref(),
+                        top_reversed_requirement,
+                        || window_plan.equivalence_properties(),
+                    )
+                }
+            } else {
+                requirements_compatible(
+                    top_reversed_requirement,
+                    window_plan.required_input_ordering()[0].as_deref(),
+                    || window_plan.equivalence_properties(),
+                ) || requirements_compatible(
+                    window_plan.required_input_ordering()[0].as_deref(),
+                    top_reversed_requirement,
+                    || window_plan.equivalence_properties(),
+                )
+            }
+        })
+        .collect::<Vec<_>>();
+
+    flags.iter().all(|o| *o)
+}
+
+fn should_reverse_window_exec(
+    required: Option<&[PhysicalSortRequirements]>,
+    request_ordering: Option<&[PhysicalSortRequirements]>,
+    input_schema: &SchemaRef,
+) -> bool {
+    match (required, request_ordering) {
+        (_, None) => false,
+        (None, Some(_)) => false,
+        (Some(required), Some(request_ordering)) => {
+            if required.len() > request_ordering.len() {
+                return false;
+            }
+            let alignment_flags = required
+                .iter()
+                .zip(request_ordering.iter())
+                .filter_map(|(required_expr, request_expr)| {
+                    // Only check the alignment of non-partition columns
+                    if request_expr.sort_options.is_some()
+                        && required_expr.sort_options.is_some()
+                    {
+                        Some(check_alignment(input_schema, request_expr, required_expr))
+                    } else if request_expr.expr.eq(&required_expr.expr) {
+                        None
+                    } else {
+                        Some(false)
+                    }
+                })
+                .collect::<Vec<_>>();
+            if alignment_flags.is_empty() {
+                false
+            } else {
+                alignment_flags.iter().all(|o| *o)
+            }
+        }
+    }
+}
+
+fn extract_window_info_from_plan(
+    plan: &Arc<dyn ExecutionPlan>,
+) -> Option<WindowExecInfo> {
```

Comment:
I think this function can return `Result<WindowExecInfo>`. This change would remove some the `unwrap`s.

Reply:
Sure